src/main/java/racingcargame/RacingCarGame.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * RacingCarGame
+ *
+ * version 1.0
+ *
+ * 2021-11-09
+ *
+ * 저작권 nextstep
+ */
+
+package racingcargame;
+
+import racingcargame.domain.RacingCar;
+import racingcargame.ui.InputView;
+import racingcargame.ui.ResultView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RacingCarGame {
+    private final InputView inputView;
+    private final ResultView resultView;
+
+    private int carCount;
+    private int tryCount;
+
+    private List<RacingCar> racingCars = new ArrayList<>();;
+
+    public RacingCarGame(InputView inputView, ResultView resultView) {
+        this.inputView = inputView;
+        this.resultView = resultView;
+    }
+
+    public void start() {
+        readInput();
+        prepareRacingCars();
+        moveAndPrintResult();
+    }
+
+    private void readInput() {
+        carCount = inputView.readCarCount();
+        tryCount = inputView.readTryCount();
+    }
+
+    protected void prepareRacingCars() {
+        racingCars.clear();
+        while (carCount-- > 0) {
+            racingCars.add(new RacingCar());
+        }
+    }
+
+    protected void moveAndPrintResult() {
+        resultView.printResultMessage();
+        while (tryCount-- > 0) {
+            moveAndPrintState();
+        }
+    }
+
+    private void moveAndPrintState() {
+        racingCars.stream().forEach(r -> {
+            r.move();
+            resultView.printCurrentStateOfRacingCar(r);
+        });
+        resultView.print("");
```

Comment:
```suggestion
        racingCars.forEach(racingCar -> {
            racingCar.move();
            resultView.printCurrentStateOfRacingCar(racingCar);
        });
        resultView.print("");
```

stream 없이 가능하고 변수명도 좀 더 명확하게 지어주는 것은 어떨까요?

Reply:
stream에서만 forEach가 가능한 줄 알았는데, Iterable에도 forEach메소드가 있었군요.😖 향상된 for로 동작한다는 것도 확인할 수 있었습니다. 감사합니다!

변수명은 racingCars.forEach()라서 앞글자 r만 따서 써도 의미를 알 수 있을 거라 생각해서 한글자로 썼던건데, 명확하게 써주는게 더 보기 편할까요?

Reply:
넵 의미 있는 이름이 더 읽기 좋지 않을까요? 🙂